redfish/provider/resource_redfish_virtual_media.go

```diff
@@ -1,26 +1,456 @@
 package provider
 
-// import (
-// 	"context"
-// 	"fmt"
-// 	"strings"
-
-// 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-// 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-// 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
-// 	"github.com/stmcginnis/gofish"
-// 	"github.com/stmcginnis/gofish/redfish"
-// )
-
-// func resourceRedfishVirtualMedia() *schema.Resource {
-// 	return &schema.Resource{
-// 		CreateContext: resourceRedfishVirtualMediaCreate,
-// 		ReadContext:   resourceRedfishVirtualMediaRead,
-// 		UpdateContext: resourceRedfishVirtualMediaUpdate,
-// 		DeleteContext: resourceRedfishVirtualMediaDelete,
-// 		Schema:        getResourceRedfishVirtualMediaSchema(),
-// 	}
-// }
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"terraform-provider-redfish/redfish/models"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &virtualMediaResource{}
+)
+
+// NewVirtualMediaResource is a helper function to simplify the provider implementation.
+func NewVirtualMediaResource() resource.Resource {
+	return &virtualMediaResource{}
+}
+
+// virtualMediaResource is the resource implementation.
+type virtualMediaResource struct {
+	p *redfishProvider
+}
+
+const (
+	serviceError = "service error"
```

Comment:
There's a new file constants.go where these common error messages are present, you can re-use the same and add new error messages there only

Reply:
done